src/test/java/seedu/address/logic/commands/SwitchParserModeCommandTest.java

```diff
@@ -0,0 +1,91 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.buyer.TypicalBuyers.getTypicalBuyerList;
+import static seedu.address.testutil.meetup.TypicalMeetUps.getTypicalMeetUpList;
+import static seedu.address.testutil.property.TypicalProperties.getTypicalPropertyList;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.parser.ParserMode;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code SwitchParserModeCommand}.
+ */
+public class SwitchParserModeCommandTest {
+
+    private Model model = new ModelManager(getTypicalBuyerList(), new UserPrefs(), getTypicalMeetUpList(),
+            getTypicalPropertyList());
+
+    @Test
+    public void execute_validParserModeSwitch_success() {
+        SwitchParserModeCommand switchParserModeCommandBuyer = new SwitchParserModeCommand(ParserMode.BUYER);
```

Comment:
Good logic implementation for parser switching, however do leave documentation or comments so that it is clear what each block of codes does and which class Buyer/Property/MeetUp it affects.